api/src/main/java/org/pentaho/metaverse/api/analyzer/kettle/step/SubtransAnalyzer.java

```diff
@@ -80,35 +82,45 @@ public void linkUsedFieldToSubTrans( IMetaverseNode originalFieldNode, TransMeta
           // Create a new node for the step, to be used as the parent of the the field we want to link to
           IMetaverseNode subTransStepNode = stepAnalyzer.createNodeFromDescriptor( stepDescriptor );
 
-          try {
-            RowMetaInterface rowMetaInterface = rfrm.getParentStepMeta().getParentTransMeta().getStepFields( step );
-            for ( int i = 0; i < rowMetaInterface.getFieldNames().length; i++ ) {
-              String field = rowMetaInterface.getFieldNames()[ i ];
-              if ( fieldPredicate.test( field ) ) {
-                // Create the descriptor for the trans field that is derived from the incoming result field
-                IComponentDescriptor stepFieldDescriptor = new MetaverseComponentDescriptor( field,
-                        DictionaryConst.NODE_TYPE_TRANS_FIELD, subTransStepNode, descriptor.getContext() );
-
-                // Create the node
-                IMetaverseNode subTransField =
-                        stepAnalyzer.createFieldNode( stepFieldDescriptor, rowMetaInterface.getValueMeta( i ), step.getName(), false );
-
-                // Add the link
-                stepAnalyzer.getMetaverseBuilder().addLink( originalFieldNode, DictionaryConst.LINK_DERIVES, subTransField );
-
-                // no need to keep looking for a match on field name, we just handled it.
-                return;
-              }
-            }
-          } catch ( KettleStepException e ) {
-            log.warn( Messages.getString( "WARN.SubtransAnalyzer.RowsFromResultNotFound", subTransMeta.getName() ),
-                    e );
+          if ( linkUsedFieldToStepField( originalFieldNode, subTransMeta, descriptor, fieldPredicate, step, rfrm,
+            subTransStepNode ) ) {
+            return;
           }
         }
       }
     }
   }
 
+  private boolean linkUsedFieldToStepField( IMetaverseNode originalFieldNode, TransMeta subTransMeta,
```

Comment:
it's not obvious to me what the boolean return value represents.  can we either add javadoc or rename method so it's clear?

Reply:
Added javadoc comment